src/stack.ts

```diff
@@ -11,172 +11,252 @@
+ * A Pulumi CDK App component. This is the entrypoint to your Pulumi CDK application.
+ * The second argument is a callback function where all CDK resources must be created.
+ *
+ * @example
+ * import * as s3 from 'aws-cdk-lib/aws-s3';
+ * import * as pulumicdk from '@pulumi/cdk';
+ *
+ * const app = new pulumicdk.App('app', (scope: pulumicdk.App): pulumicdk.AppOutputs => {
+ *   // All resources must be created within a Pulumi Stack
+ *   const stack = new pulumicdk.Stack(scope, 'pulumi-stack');
+ *   const bucket = new s3.Bucket(stack, 'my-bucket');
+ *   return {
+ *     bucket: stack.asOutput(bucket.bucketName),
+ *   };
+ * });
+ *
+ * export const bucket = app.outputs['bucket'];
  */
-class StackComponent extends pulumi.ComponentResource implements StackComponentResource {
-    /** @internal */
-    name: string;
+export class App
+    extends pulumi.ComponentResource<AppResource>
+    implements ICloudAssemblyDirectoryProducer, AppComponent
+{
+    public readonly name: string;
+
+    /**
+     * The collection of outputs from the AWS CDK Stack represented as Pulumi Outputs.
+     * Each CfnOutput defined in the AWS CDK Stack will populate a value in the outputs.
+     */
+    public outputs: { [outputId: string]: pulumi.Output<any> } = {};
+
+    /**
+     * @internal
+     */
+    public readonly component: pulumi.ComponentResource;
+
+    /**
+     * @internal
+     */
+    public readonly stacks: { [artifactId: string]: cdk.Stack } = {};
 
     /** @internal */
-    converter: AppConverter;
+    public converter: Promise<AppConverter>;
 
     /**
      * @internal
      */
-    readonly component: pulumi.ComponentResource;
+    public appOptions?: AppOptions;
 
     /**
      * The directory to which cdk synthesizes the CloudAssembly
      * @internal
      */
-    public assemblyDir: string;
+    public assemblyDir!: string;
 
     /**
-     * Any stack options that are supplied by the user
      * @internal
      */
-    public options?: StackOptions;
+    readonly dependencies: CdkConstruct[] = [];
+
+    private readonly createFunc: (scope: App) => AppOutputs | void;
+    private _app?: cdk.App;
+    private appProps?: cdk.AppProps;
+
+    constructor(id: string, createFunc: (scope: App) => void | AppOutputs, props?: AppResourceOptions) {
+        super('cdk:index:App', id, props?.appOptions, props);
+        this.appOptions = props?.appOptions;
+        this.createFunc = createFunc;
+        this.component = this;
+
+        this.name = id;
+        this.appProps = props?.appOptions?.props;
+        const data = this.getData();
+        this.converter = data.then((d) => d.converter);
+
+        // This grabs the outputs off of the stacks themselves after they
+        // have been converted. This allows us to present the outputs property
+        // as a plain value instead of an Output value.
+        const outputs = this.converter.then((converter) => {
```

Comment:
This allows us to have `Record<string, Output<any>>` instead of `Output<Record<string, any>>`, right?

Reply:
Yes!